lib/api/controllers/cli/cleanDb.js

```diff
@@ -19,6 +19,94 @@
  * limitations under the License.
  */
 
+const 
+  Bluebird = require('bluebird'),
+  Request = require('kuzzle-common-objects').Request;
+
 module.exports = function cliResetStorage (kuzzle) {
-  return () => kuzzle.resetStorage();
+  return (request, statusUpdate) => resetStorage(kuzzle, statusUpdate);
 };
+
+/**
+ * Flushes the internal storage components (internalEngine index, cache and memory storage)
+ *
+ * @param {Kuzzle} kuzzle 
+ * @param {Function} statusUpdate 
+ * @returns Promise
+ */
+function resetStorage (kuzzle, statusUpdate) {
+  statusUpdate('Kuzzle reset initiated: this may take a while...');
+  
+  return deleteUsers(kuzzle, statusUpdate)
+    .then(() => kuzzle.internalEngine.deleteIndex())
+    .then(() => statusUpdate('Kuzzle internal database deleted'))
+    .then(() => {
+      const promises = ['internalCache', 'memoryStorage']
```

Comment:
I would be more confortable if we only flushed the `internalCache` as we do for ES data, `memoryStorage` should not be flushed.